### PR TITLE
chore: updated a11y texts for oval fitting, updated error text to men…

### PR DESCRIPTION
**Pull request by thaddmt**
…tion camera's center

#### Description of changes
- Updated a11y text to read out when face is outside the oval that it should move to the center of the oval
- Added some a11y text about percentage of match bar filled, only says 50% and 100% because more percntages would be too wordy
- Updated error text to mention center of the camera

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .changeset/pink-tools-jog.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@aws-amplify/ui-react-liveness": patch
+---
+
+chore: updated a11y texts for oval fitting, updated error text to men…
```
Comment (ioanabrooks) [inline .changeset/pink-tools-jog.md:5]:
```suggestion
chore: updated a11y texts for oval fitting, updated error text to mention camera's center
```

File: packages/react-liveness/src/components/FaceLivenessDetector/service/utils/__tests__/liveness.test.ts [hunk 2]
```diff
@@ -54,10 +54,44 @@ describe('Liveness Helper', () => {
...
+      const sessionInfo = mockSessionInformation;
+
+      const { faceMatchState } = getFaceMatchStateInLivenessOval(
+        face,
+        ovalDetails,
+        initialIntersection,
+        sessionInfo
+      );
+
+      expect(faceMatchState).toBe(FaceMatchState.TOO_CLOSE);
```
Comment (ioanabrooks) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/utils/__tests__/liveness.test.ts:92]: Test reads it should return Too Far but expecting too close here?

Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/utils/__tests__/liveness.test.ts:92]: Will fix, copy paste mistake

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx [hunk 3]
```diff
@@ -149,16 +153,34 @@ export const Hint: React.FC<HintProps> = ({ hintDisplayText }) => {
...
+      let a11yHintString = resultHintString;
+      if (faceMatchState === FaceMatchState.OFF_CENTER) {
+        a11yHintString = FaceMatchStateStringMap[faceMatchState];
+      }
+
+      // If the face match percentage reaches a 50% or 100% marks append it to the a11y label
+      if (faceMatchState === FaceMatchState.MATCHED) {
+        a11yHintString = `${a11yHintString}. 100% face fit.`;
+      } else if (faceMatchPercentage! > 50) {
+        a11yHintString = `${a11yHintString}. 50% face fit.`;
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx:166]: Do these these need to be included in `displayText` for i18n?

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx [hunk 4]
```diff
@@ -149,16 +153,34 @@ export const Hint: React.FC<HintProps> = ({ hintDisplayText }) => {
...
           variation={
             faceMatchState === FaceMatchState.TOO_CLOSE ? 'error' : 'primary'
           }
         >
-          <View aria-live="assertive" aria-label={resultHintString}>
-            {resultHintString}
+          <View
+            aria-live="assertive"
+            aria-label={a11yHintString}
+            className="amplify-visually-hidden"
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx:179]: Can we use `VisuallyHidden` here?

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx (hunk 3 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx:166]: good call out, will add

File: packages/react-liveness/src/components/FaceLivenessDetector/displayText.ts [hunk 5]
```diff
@@ -12,6 +12,8 @@ export type HintDisplayText = {
   hintIlluminationNormalText?: string;
   hintHoldFaceForFreshnessText?: string;
   hintCenterFaceText?: string;
+  hintFaceOffCenterText?: string;
+  hintMatchIndicatorA11yText?: string;
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/displayText.ts:16]: `A11y` is not always a known acronym, think we can just remove it here:

```suggestion
  hintMatchIndicatorText?: string;
```

File: packages/react-liveness/src/components/FaceLivenessDetector/displayText.ts [hunk 6]
```diff
@@ -96,6 +100,7 @@ export const defaultLivenessDisplayText: Required<LivenessDisplayText> = {
   hintIlluminationTooDarkText: 'Move to brighter area',
   hintIlluminationNormalText: 'Lighting conditions normal',
   hintHoldFaceForFreshnessText: 'Hold still',
+  hintMatchIndicatorA11yText: 'face fit.',
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/displayText.ts:103]: See above comment on line 16:
```suggestion
  hintMatchIndicatorText: 'face fit.',
```

File: packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts [hunk 7]
```diff
@@ -381,6 +381,11 @@ export function getFaceMatchStateInLivenessOval(
     Math.abs(maxOvalY - maxFaceY) < ovalMatchHeightThreshold
   ) {
     faceMatchState = FaceMatchState.MATCHED;
+  } else if (
+    (minOvalX > minFaceX && maxOvalX > maxFaceX) ||
+    (minFaceX > minOvalX && maxFaceX > maxOvalX)
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts:386]: Can we move these to variables for future readability

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx [hunk 8]
```diff
@@ -149,16 +153,30 @@ export const Hint: React.FC<HintProps> = ({ hintDisplayText }) => {
         resultHintString = FaceMatchStateStringMap[faceMatchState];
       }
 
+      // If the face is outside the oval set the aria-label to a string about centering face in oval
+      let a11yHintString = resultHintString;
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx:157]: We can just continue mutating `resultHintString` here

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx [hunk 9]
```diff
@@ -149,16 +153,30 @@ export const Hint: React.FC<HintProps> = ({ hintDisplayText }) => {
...
+      if (faceMatchState === FaceMatchState.OFF_CENTER) {
+        a11yHintString = FaceMatchStateStringMap[faceMatchState];
+      }
+
+      // If the face match percentage reaches a 50% or 100% marks append it to the a11y label
+      if (faceMatchState === FaceMatchState.MATCHED) {
+        a11yHintString = `${a11yHintString}. 100% ${hintDisplayText.hintMatchIndicatorA11yText}`;
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx:164]: Think this needs more consideration for `displayText` overrides, users may not want the hardocded "%" and other english based formatting

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx (hunk 8 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx:157]: We have separate variables here because there are some cases where we want the a11y text to read something different than what is on screen. So in some cases on screen it will say `move closer` but for the a11y text it will read something different like `face is outside oval, center your face`

Comment (calebpollman) [reply packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx:157]: Ahhhh, missed that `resultHintString` was still being used here

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx [hunk 10]
```diff
@@ -30,16 +30,18 @@ export const selectFaceMatchStateBeforeStart = createLivenessSelector(
...
   hintDisplayText: Required<HintDisplayText>;
 }
 
 const defaultToast = (text: string, isInitial = false) => {
   return (
     <Toast size="large" variation="primary" isInitial={isInitial}>
-      <View aria-live="assertive" aria-label={text}>
-        {text}
-      </View>
+      <View aria-live="assertive">{text}</View>
```
Comment (thaddmt) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx:44]: removed various aria-labels where there was already an aria-live attribute because osx voice over would double say the aria live and aria label which was extremely annoying

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx [hunk 11]
```diff
@@ -90,7 +94,14 @@ export const Hint: React.FC<HintProps> = ({ hintDisplayText }) => {
 
   const getInstructionContent = () => {
     if (isStartView) {
-      return defaultToast(hintDisplayText.hintCenterFaceText, true);
+      return (
+        <>
+          <VisuallyHidden role="alert">
+            {hintDisplayText.hintCenterFaceInstructionText}
+          </VisuallyHidden>
+          {defaultToast(hintDisplayText.hintCenterFaceText, true)}
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx:102]: Curious if there is reasoning to use an imperative API here? As far as i can tell `defaultToast` can just be a React element

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx [hunk 12]
```diff
@@ -149,16 +163,31 @@ export const Hint: React.FC<HintProps> = ({ hintDisplayText }) => {
...
+        a11yHintString = FaceMatchStateStringMap[faceMatchState];
+      }
+
+      // If the face match percentage reaches 50% append it to the a11y label
+      if (
+        faceMatchState === FaceMatchState.TOO_FAR &&
+        faceMatchPercentage! > 50
+      ) {
+        a11yHintString = hintDisplayText.hintMatchIndicatorText;
+      }
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx:178]: Is it possible for both these cases to be truthy?

**Review by calebpollman — COMMENTED** [review]
LGTM, just left a couple questions

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx (hunk 12 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx:178]: Don't think so at this time since those are distinctly different faceMatchStates

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx (hunk 11 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx:102]: No particular reason at this time, can update

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx (hunk 12 above)
Comment (calebpollman) [reply packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx:187]: Cool, that's how i was reading it. Not a blocker but might make sense to put this in a util and/or update to an `if`/`else`

**Review by calebpollman — DISMISSED** [review]

**Review by calebpollman — APPROVED** [review]

**Review by esauerbo — APPROVED** [review]
